src/CheckRightsService.Broker/Consumers/AccessCollectionValidatorConsumer.cs

```diff
@@ -0,0 +1,39 @@
+﻿using LT.DigitalOffice.CheckRightsService.Data.Interfaces;
+using LT.DigitalOffice.Kernel.AccessValidatorEngine.Requests;
+using LT.DigitalOffice.Kernel.Broker;
+using MassTransit;
+using Microsoft.AspNetCore.Mvc;
+using System;
+using System.Threading.Tasks;
+
+namespace LT.DigitalOffice.CheckRightsService.Broker.Consumers
+{
+    public class AccessCollectionValidatorConsumer : IConsumer<IAccessValidatorCheckRightsCollectionServiceRequest>
+    {
+        private readonly ICheckRightsRepository _repository;
+
+        public AccessCollectionValidatorConsumer([FromServices] ICheckRightsRepository repository)
+        {
+            _repository = repository;
+        }
+
+        public async Task Consume(ConsumeContext<IAccessValidatorCheckRightsCollectionServiceRequest> context)
+        {
+            var response = OperationResultWrapper.CreateResponse(HasRights, context.Message);
+            await context.RespondAsync<IOperationResult<bool>>(response);
+        }
+
+        private object HasRights(IAccessValidatorCheckRightsCollectionServiceRequest request)
+        {
+            foreach(var rigthId in request.RightIds)
+            {
+                if (!_repository.IsUserHasRight(request.UserId, rigthId))
+                {
+                    throw new Exception("Such user doesn't exist or does not have this rights.");
+                }
```

Comment:
Does not make much sense to throw error, because `repository.IsUserHasRight` return boolean variable  and it must be definitely put something in message body of response broker

Reply:
I changed the return value to false and adjusted the tests